### PR TITLE
hmm rng states start at 1, not 0.

### PR DESCRIPTION
**Pull request by charlesm93**
## Summary

Solves issue#2112. The states returned by hmm_hidden_state_rng now live on {1, 2, ..., K}, rather than {0, 1, ..., K - 1}.

## Tests

The existing unit tests were adjusted to handle state values that start at 1.

## Release notes

The states returned by hmm_hidden_state_rng now live on {1, 2, ..., K}, rather than {0, 1, ..., K - 1}.

## Checklist

- [x] Math issue #2112 

- [x] Copyright holder: Charles Margossian and Columbia University.

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [ ] the basic tests are passing

    - [x] unit tests pass (to run, use: `./runTests.py test/unit/math/prim/prob/hmm_latent_rng_test.cpp`)
    - [ ] header checks pass, (`make test-headers`)
    - [ ] dependencies checks pass, (`make test-math-dependencies`)
    - [ ] docs build, (`make doxygen`)
    - [x] code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

File: test/unit/math/prim/prob/hmm_latent_rng_test.cpp
Comment (bob-carpenter) [inline test/unit/math/prim/prob/hmm_latent_rng_test.cpp:17]: "male" -> "make"

**Review by bob-carpenter — APPROVED** [review]
Looks good.  Thanks for the quick fix!  The doc typo correction is optional, so I'll just approve this.  If you want to correct it, I can approve again quickly.

File: stan/math/prim/prob/hmm_latent_rng.hpp
Comment (syclik) [inline stan/math/prim/prob/hmm_latent_rng.hpp:73]: @charlesm93, there's a terribly named variable that stores the indexing. If you wanted to be proper about this, you could change this (and other instances) of `+ 1` with `+ stan::error_index::value`. https://github.com/stan-dev/math/blob/develop/stan/math/prim/meta/error_index.hpp

It's not only terribly named... it has no doc and I don't know how you would have known about it. I'll create issues to tackle that later.

The intention was if someone really wanted to 0-index things (like if it were from a C++ function), they'd compile with `-DERROR_INDEX=0` and that would magically make things like this, bounds checks on arrays, and other places where there is indexing, be 0-indexed. By default, it's 1-indexed.

Alternately, we can just ditch the generality and explicitly say that everything is 1-indexed, as confusing as it is with C++ being 0-indexed, and make life simpler.

Anyway, just adding a comment so you know that there's a variable that defines the indexing. If you want to switch it, great. If not, also great.

Comment (bbbales2) [reply stan/math/prim/prob/hmm_latent_rng.hpp:73]: @charlesm93 I think it makes sense to do what @syclik is saying here.

I think how this works is you just add/subtract `stan::error_index::value` everywhere before you added/subtracted `1` (including the tests). It's all tested in 1-indexing mode (how Stan uses it) and then if someone wants the 0-indexing they can do the make flag.

Release is the 19th and code freeze is the 12th for reference.

Comment (bbbales2) [reply stan/math/prim/prob/hmm_latent_rng.hpp]: @charlesm93 since this was sitting here and it was a relatively simple but confusing change I just did it (and made Bob's edit). Have a look and if you approve we can merge this.

**Comment by bbbales2** [comment]
Got this error on the Mac integration tests ([log](https://jenkins.mc-stan.org/blue/organizations/jenkins/Stan/detail/downstream_tests/2044/pipeline)):

```
bin/stanc src/test/test-models/good/array-expr/validate_array_expr_big.stan --o=test/test-models/good/array-expr/validate_array_expr_big.hpp
bin/stanc: line 1: Not: command not found
make: *** [test/test-models/good/array-expr/validate_array_expr_big.hpp] Error 127
make: *** Waiting for unfinished jobs....
```

I'm not sure what it's about. Just restarting the integration tests.

**Comment by bbbales2** [comment]
The same error popped up again. Do you recognize this one @rok-cesnovar?

Log here: https://jenkins.mc-stan.org/blue/organizations/jenkins/Stan/detail/downstream_tests/2044/pipeline

```
bin/stanc src/test/test-models/good/array-expr/validate_array_expr_big.stan --o=test/test-models/good/array-expr/validate_array_expr_big.hpp
bin/stanc: line 1: Not: command not found
make: *** [test/test-models/good/array-expr/validate_array_expr_big.hpp] Error 127
make: *** Waiting for unfinished jobs....
```

**Comment by rok-cesnovar** [comment]
This looks as if the stanc download failed and did not retry. Restart it and I will make an issue to make this more robust.

**Comment by rok-cesnovar** [comment]
I restarted the tests, will go green once they pass.

File: stan/math/prim/prob/hmm_latent_rng.hpp
Comment (charlesm93) [reply stan/math/prim/prob/hmm_latent_rng.hpp]: @bbbales2 Thanks for jumping in. I agree with what @syclik proposes. I'll take a quick look.

Comment (charlesm93) [reply stan/math/prim/prob/hmm_latent_rng.hpp]: @bbbales2 you missed one +1, I translated it into an index error. Once the integration tests pass, I'll merge it (small caprice, I love pushing the merge button!)

**Comment by charlesm93** [comment]
All the comments by @bob-carpenter and @syclik have been addressed. Can one of the reviewers approve the PR?

**Review by bob-carpenter — APPROVED** [review]
Thanks for updating!

**Pull request merged**
